BLSQ/iaso
# Polio 550 drag and drop component

The exiting `FileComponent` allowed drag and dropping files, but it wasn't clear to the user.

## Self proof reading checklist



## Changes

- Created `SimpleModal`, `ConfirmCancelButtons`, `ConfirmCancelModal`  to replace `ConfirmCancelDialogComponent` and be able to make other types of modals (by composing `SimpleModal`). The main point was to remove `renderTrigger` so that when we conditionally render the modal component, the API calls are not made unless the modal is rendered.
- Created a `makeFullModal` HOC to facilitate making modals with `ConfirmCancelModal`
- Created `CustomInput` based on `OrgUnitTreeviewPicker`
- Added dependency to [react-dropzone](https://react-dropzone.js.org)
- Used `CustomInput` and react-dropzone to make `FileUpload` component: users can drag and drop files into the component. OnDragOver, it will show a dropzone. Users can also click on the input to open the file selection window
- Refactor `CreateEditBudgetEvent` to use `ConfirmCancelModal`, `makFullModal` and `FileUpload`



## How to test

- Go to a campaign's budget
- Click add step
- click on the "Files" input to open file selection window
- then drag and drop another selection of files in the component

To test the refactor:
- Open the modal to create a new step
- Fill it in
- Block `/api/polio/budgetfiles` and `/api/polio/confirm_budget`
- Confirm
- The table should show an additional icon
- click on it
- You should be able to resend files (but you have to re-authorize the endpoints first for it to work)

## Print screen / video

https://user-images.githubusercontent.com/38907762/189665826-73b498b2-ef85-4fc5-beb5-53a4336fa600.mov

## Notes

`SimpleModal`, `ConfirmCancelButtons`, `ConfirmCancelModal`,`makeFullModal` and `FileUpload` should be moved to bluesquare-components. I suggest to create a ticket after this PR is merged
